File: src/main/java/com/sequenceiq/periscope/rest/converter/ClusterConverter.java
package com.sequenceiq.periscope.rest.converter;

import org.springframework.stereotype.Component;

import com.sequenceiq.periscope.domain.Cluster;
import com.sequenceiq.periscope.rest.json.ClusterJson;

@Component
public class ClusterConverter extends AbstractConverter<ClusterJson, Cluster> {

    @Override
    public ClusterJson convert(Cluster source) {
        ClusterJson json = new ClusterJson();
        json.setId(source.getId());
        json.setHost(source.getHost());
        json.setPort(source.getPort());
        json.setState(source.getState().name());
        json.setAppMovement(source.isAppMovementAllowed() ? "allowed" : "prohibited");
        return json;
    }

}

File: autoscale/src/main/java/com/sequenceiq/periscope/rest/converter/ClusterConverter.kt
package com.sequenceiq.periscope.rest.converter

import org.springframework.stereotype.Component

import com.sequenceiq.periscope.domain.Cluster
import com.sequenceiq.periscope.api.model.ClusterJson

@Component
class ClusterConverter : AbstractConverter<ClusterJson, Cluster>() {

    override fun convert(source: Cluster): ClusterJson {
        val json = ClusterJson()
        json.id = source.id
        json.stackId = source.stackId
        json.host = source.host
        json.port = source.port
        json.state = source.state.name
        return json
    }

}
